external-crates/move/move-core/types/src/annotated_value.rs

```diff
@@ -0,0 +1,424 @@
+        self,
+        deserializer: D,
+    ) -> Result<Self::Value, D::Error> {
+        Ok((self.name.clone(), self.layout.deserialize(deserializer)?))
+    }
+}
+
+impl<'d> serde::de::DeserializeSeed<'d> for &MoveStructLayout {
+    type Value = MoveStruct;
+
+    fn deserialize<D: serde::de::Deserializer<'d>>(
+        self,
+        deserializer: D,
+    ) -> Result<Self::Value, D::Error> {
+        let fields = deserializer
+            .deserialize_tuple(self.fields.len(), DecoratedStructFieldVisitor(&self.fields))?;
+        Ok(MoveStruct {
+            type_: self.type_.clone(),
+            fields,
+        })
+    }
+}
+
+impl serde::Serialize for MoveValue {
+    fn serialize<S: serde::Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        match self {
+            MoveValue::Struct(s) => s.serialize(serializer),
+            MoveValue::Bool(b) => serializer.serialize_bool(*b),
+            MoveValue::U8(i) => serializer.serialize_u8(*i),
+            MoveValue::U16(i) => serializer.serialize_u16(*i),
+            MoveValue::U32(i) => serializer.serialize_u32(*i),
+            MoveValue::U64(i) => serializer.serialize_u64(*i),
+            MoveValue::U128(i) => serializer.serialize_u128(*i),
+            MoveValue::U256(i) => i.serialize(serializer),
+            MoveValue::Address(a) => a.serialize(serializer),
+            MoveValue::Signer(a) => a.serialize(serializer),
+            MoveValue::Vector(v) => {
+                let mut t = serializer.serialize_seq(Some(v.len()))?;
+                for val in v {
+                    t.serialize_element(val)?;
+                }
+                t.end()
+            }
+        }
+    }
+}
+
+struct MoveFields<'a>(&'a [(Identifier, MoveValue)]);
+
+impl<'a> serde::Serialize for MoveFields<'a> {
+    fn serialize<S: serde::Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        let mut t = serializer.serialize_map(Some(self.0.len()))?;
+        for (f, v) in self.0.iter() {
+            t.serialize_entry(f, v)?;
+        }
+        t.end()
+    }
+}
+
+impl serde::Serialize for MoveStruct {
+    fn serialize<S: serde::Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        // Serialize a Move struct as Serde struct type named `struct `with two fields named `type` and `fields`.
+        // `fields` will get serialized as a Serde map.
+        // Unfortunately, we can't serialize this in the logical way: as a Serde struct named `type` with a field for
+        // each of `fields` because serde insists that struct and field names be `'static &str`'s
+        let mut t = serializer.serialize_struct(MOVE_STRUCT_NAME, 2)?;
+        // serialize type as string (e.g., 0x0::ModuleName::StructName<TypeArg1,TypeArg2>) instead of (e.g.
+        // { address: 0x0...0, module: ModuleName, name: StructName, type_args: [TypeArg1, TypeArg2]})
+        t.serialize_field(MOVE_STRUCT_TYPE, &self.type_.to_string())?;
```

Comment:
Shouldn't this be removed?

Reply:
I don't believe so. This is serializing a struct like `0xA::M::S { f: 0 }` in Move to something like
```json
{
  "type": "0xA::M::S",
  "fields": {
     "f": 0
   }
}
```
which is also consistent with how `WithTypes` was serialized previously.

Reply:
But... there is no way anyone used this right? because deserialize did not read the type?

Reply:
We only know there was no round-tripping of them since that would fail. But the serialization is (IMO) correct.

I think keeping this output format is useful e.g., for json and the like, so I have a preference for keeping it, but I can look into what removing it  might entail (and how possible it might/might not be).